### PR TITLE
Discard samples

### LINKED ISSUE #2: Script to queue plate to discard
This script will be called from the LIMS web interface through a trigger providing a process id
it will:
 1. find all the samples going through that step
 2. find all the artifacts (Analytes) associated with those samples
 3. find all the containers associated with those artifacts
 4. finds all the artifacts associated with these containers not associated with samples in 1.
 5. go through each of the containers in 3 and route their artifacts if the root artifacts of all those artifacts have been through "Discard Sample" workflow

We need to avoid making too many queries to the LIMS so as much as possible, gather all the ids then make a search query then fetch the object using batch.

### PR DESCRIPTION
**Pull request by tcezard**
closes #2 
This script will take a step uri, find all the sample and route the plate associated with their derived artifact if all the artifact they contained have been through the Discard sample step

### CODE REVIEW

File: EPPs/prepare_discard_plate.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    '-LQC',
+    '-DQC1',
+    '-DIL1',
+    '-DIL2',
+    '-QPCR',
+    '-CST'
+]
+discard_wf_name="Discard Plates EG 1.0"
+sample_discard_wf_stage_name = "Find Derived Sample Plates EG 1.0"
+plate_discard_wf_stage_name="Discard Plates EG 1.0"
```
Comment (tcezard) [inline EPPs/prepare_discard_plate.py]: @EGCGDanForth 
Have you created these step in the dev instance?

**Comment by EGCGDanForth** [comment]
Not created them yet. I can do so now. We just need the barebones of  a workflow so won't taken long.

**Comment by EGCGDanForth** [comment]
Discard Plates EG 1.0 workflow and protocol now present in dev. Let me know if there any issues with it.

**Comment by tcezard** [comment]
It all seems to work

**Pull request merged**